src/routes/safe/components/Apps/hooks/useAppList.ts

```diff
@@ -4,27 +4,42 @@ import { APPS_STORAGE_KEY, getAppInfoFromUrl, getAppsList, getEmptySafeApp } fro
 import { AppData } from '../api/fetchSafeAppsList'
 import { SafeApp, StoredSafeApp, SAFE_APP_FETCH_STATUS } from '../types'
 import { getNetworkId } from 'src/config'
+import enqueueSnackbar from 'src/logic/notifications/store/actions/enqueueSnackbar'
+import { NOTIFICATIONS } from 'src/logic/notifications'
+import { useDispatch } from 'react-redux'
 
 type UseAppListReturnType = {
   appList: SafeApp[]
   removeApp: (appUrl: string) => void
   staticAppsList: AppData[]
+  isLoading: boolean
 }
 
-const useAppList = (): UseAppListReturnType => {
+const useAppList = (showError: boolean): UseAppListReturnType => {
   const [appList, setAppList] = useState<SafeApp[]>([])
   const [staticAppsList, setStaticAppsList] = useState<AppData[]>([])
+  const dispatch = useDispatch()
+  const [isLoading, setIsLoading] = useState(false)
 
   useEffect(() => {
     const loadAppsList = async () => {
-      const remoteAppsList = await getAppsList()
-      setStaticAppsList(remoteAppsList)
+      setIsLoading(true)
+      let result
+      try {
+        result = await getAppsList()
+      } catch (err) {
+        if (showError) {
+          dispatch(enqueueSnackbar(NOTIFICATIONS.SAFE_APPS_FETCH_MSG))
+        }
+      }
+      setStaticAppsList(result && result?.length ? result : staticAppsList)
```

Comment:
You don't need to write to the state if an error was caught

Reply:
I thought this was the way to show a notification to the user.

Reply:
I'm talking about this line:
```
setStaticAppsList(result && result?.length ? result : staticAppsList)
```

We don't need to call it if there's an error, you can move it to `try`  block